libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -135,6 +139,33 @@ public static String getSHA256Hash(String privateKey) {
         return hashedString;
     }
 
+    /**
+     * Generate a RSA-2048 keypair and get the encoded public key string
+     *
+     * @return RSA-2048 public key string
+     */
+    @Nullable
+    public static synchronized String getRSA2048PublicString(String name) {
```

Comment:
Why does there need to be an API for a specific key length?  We wouldn't this just be `getRSAPublicString(name, keyLength)`?

Reply:
Where does the private key go?  How is the public key useful without its counterpart in existence somewhere?

Reply:
I agree with @khawkins. Let's create `getRSAPublicString(name, keyLength)` instead.

Reply:
Yeah I can make the change. Currently RSA-2048 is the only usage for the push notification encryption. The private key will be stored and handled locally by the system. I will add the method to get it and make it ready for decryption.